bot/exts/utilities/color.py

```diff
@@ -0,0 +1,224 @@
+        """Create and send embed from user given colour information."""
+        r, g, b = rgb
+        name = self._rgb_to_name(rgb)
+        if name == "No match found":
+            name = None
+
+        try:
+            colour_or_color = ctx.invoked_parents[0]
+        except IndexError:
+            colour_or_color = "colour"
+
+        colour_mode = ctx.invoked_with
+        if colour_mode == "random":
+            input_colour = "random"
+        else:
+            input_colour = ctx.args[2:][0]
+
+        if colour_mode not in ("name", "hex", "random"):
+            colour_mode = colour_mode.upper()
+        else:
+            colour_mode = colour_mode.title()
+
+        desc = f"{colour_mode} information for `{name or input_colour}`."
+        colour_embed = Embed(
+            title=colour_or_color.title(),
+            description=desc,
+            colour=int(f"{r:02x}{g:02x}{b:02x}", 16)
+        )
+        colour_conversions = self.get_colour_conversions(rgb)
+        for colour_space, value in colour_conversions.items():
+            colour_embed.add_field(
+                name=colour_space,
+                value=f"`{value}`",
+                inline=True
+            )
+
+        thumbnail = Image.new("RGB", THUMBNAIL_SIZE, color=rgb)
+        buffer = BytesIO()
+        thumbnail.save(buffer, "PNG")
+        buffer.seek(0)
+        thumbnail_file = File(buffer, filename="colour.png")
+
+        colour_embed.set_thumbnail(url="attachment://colour.png")
+
+        await ctx.send(file=thumbnail_file, embed=colour_embed)
+
+    @commands.group(aliases=("color",))
+    async def colour(self, ctx: commands.Context) -> None:
+        """User initiated command to create an embed that displays colour information."""
+        if ctx.invoked_subcommand is None:
+            await invoke_help_command(ctx)
+
+    @colour.command()
+    async def rgb(self, ctx: commands.Context, red: int, green: int, blue: int) -> None:
+        """Command to create an embed from an RGB input."""
+        if any(c not in range(0, 256) for c in (red, green, blue)):
+            raise BadArgument(
+                message=f"RGB values can only be from 0 to 255. User input was: `{red, green, blue}`."
+            )
+        rgb_tuple = (red, green, blue)
+        await self.send_colour_response(ctx, rgb_tuple)
+
+    @colour.command()
+    async def hsv(self, ctx: commands.Context, hue: int, saturation: int, value: int) -> None:
+        """Command to create an embed from an HSV input."""
+        if (hue not in range(0, 361)) or any(c not in range(0, 101) for c in (saturation, value)):
+            raise BadArgument(
+                message="Hue can only be from 0 to 360. Saturation and Value can only be from 0 to 100. "
+                f"User input was: `{hue, saturation, value}`."
+            )
+        # ImageColor HSV expects S and V to be swapped
+        hsv_tuple = ImageColor.getrgb(f"hsv({hue}, {value}%, {saturation}%)")
```

Comment:
https://pillow.readthedocs.io/en/stable/reference/ImageColor.html#module-PIL.ImageColor The docs specify `hsv(hue, saturation%, value%) `
```suggestion
        hsv_tuple = ImageColor.getrgb(f"hsv({hue}, {saturation}%, {value}%)")

Reply:
This was the behavior before I swapped the values (I took the HSV values from the embed when I called `.color name Bone`):

![image](https://user-images.githubusercontent.com/42116429/139493211-b53393f4-ada5-45f9-915e-5596268631bb.png)

And the HSV values were not converted correctly.

I swapped the S and V and then got the correct embed returned:

![image](https://user-images.githubusercontent.com/42116429/139493392-b50c8a58-3303-4b1e-a716-dfafb0f5bb13.png)

Reply:
I think you're getting this because you've also swapped the order in the embed which i've also made a comment on below. Here are the correct values from an online converter
![image](https://user-images.githubusercontent.com/35849006/139496426-faeecb3e-1211-4454-bba3-b389b37491c9.png)

Reply:
Yeah, got them both swapped back, should be ok now?

Reply:
The values needed to be swapped, fixed with 8808914.